src/Compilers/Core/Portable/SourceGenerator.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Threading;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis
+{
+    public interface ISourceGenerator
+    {
+        void Execute(SourceGeneratorContext context);
+    }
+
+    public struct SourceGeneratorContext
+    {
+        public Compilation Compilation { get; }
+
+        // TODO: replace AnalyzerOptions with an differently named type that is otherwise identical.
+        // The concern being that something added to one isn't necessarily applicable to the other.
+        public AnalyzerOptions AnalyzerOptions { get; }
+
+        public CancellationToken CancellationToken { get; }
+
+        public void ReportDiagnostic(Diagnostic diagnostic) { throw new NotImplementedException(); }
+
+        public void AddSource(string fileNameHint, SourceText sourceText) { throw new NotImplementedException(); }
+    }
+
+    public interface IDependsOnAdditionalFileGenerator : ISourceGenerator
```

Comment:
We landed (for now) as interfaces being the way source generators would tag extra things they rely on. We discussed two other approaches:

1. **Attributes:** good, but often limited in that you are restricted to the types of data that you can put in them; was worrisome if we needed any future flexibility.
2. **Registration mechanism:** we had one approach where ISourceGenerator.Register() existed, and a generator could call some methods that said "I depend on these extensions". This was helpful, but then meant we had to actually run more code to figure out the result.

Reply:
Lets call these TriggeredBy, not DependsOn